audio_processing/gpt-sovits/gpt-sovits.py

```diff
@@ -0,0 +1,289 @@
+        if args.benchmark:
+            start = int(round(time.time() * 1000))
+        if args.onnx or args.onnx_vits:
+            audio1 = self.sess.run(None, {
+                "text_seq" : text_seq,
+                "pred_semantic" : pred_semantic, 
+                "ref_audio" : ref_audio
+            })
+        else:
+            audio1 = self.sess.run({
+                "text_seq" : text_seq,
+                "pred_semantic" : pred_semantic, 
+                "ref_audio" : ref_audio
+            })
+        if args.benchmark:
+            end = int(round(time.time() * 1000))
+            logger.info("\tvits processing time {} ms".format(end-start))
+        return audio1[0]
+
+
+class SSLModel():
+    def __init__(self, sess):
+        self.sess = sess
+
+    def forward(self, ref_audio_16k):
+        if args.benchmark:
+            start = int(round(time.time() * 1000))
+        if args.onnx:
+            last_hidden_state = self.sess.run(None, {
+                "ref_audio_16k" : ref_audio_16k
+            })
+        else:
+            last_hidden_state = self.sess.run({
+                "ref_audio_16k" : ref_audio_16k
+            })
+        if args.benchmark:
+            end = int(round(time.time() * 1000))
+            logger.info("\tssl processing time {} ms".format(end-start))
+        return last_hidden_state[0]
+
+
+def generate_voice(ssl, t2s_encoder, t2s_first_decoder, t2s_stage_decoder, vits):
+    gpt = T2SModel(t2s_encoder, t2s_first_decoder, t2s_stage_decoder,)
+    gpt_sovits = GptSoVits(gpt, vits)
+    ssl = SSLModel(ssl)
+
+    input_audio = args.audio
+
+    ref_phones = g2p(args.transcript)
+    ref_seq = np.array([cleaned_text_to_sequence(ref_phones)], dtype=np.int64)
+
+    text_phones = g2p(args.input)
+    text_seq = np.array([cleaned_text_to_sequence(text_phones)], dtype=np.int64)
+
+    # empty for ja or en
+    ref_bert = np.zeros((ref_seq.shape[1], 1024), dtype=np.float32)
+    text_bert = np.zeros((text_seq.shape[1], 1024), dtype=np.float32)
+    
+    vits_hps_data_sampling_rate = 32000
+
+    zero_wav = np.zeros(
+        int(vits_hps_data_sampling_rate * 0.3),
+        dtype=np.float32,
+    )
+    wav16k, sr = librosa.load(input_audio, sr=16000)
+    wav16k = np.concatenate([wav16k, zero_wav], axis=0)
+    wav16k = wav16k[np.newaxis, :]
+    ref_audio_16k = wav16k # hubertの入力のみpaddingする
+
+    wav32k, sr = librosa.load(input_audio, sr=vits_hps_data_sampling_rate)
+    wav32k = wav32k[np.newaxis, :]
+
+    ssl_content = ssl.forward(ref_audio_16k)
+
+    a = gpt_sovits.forward(ref_seq, text_seq, ref_bert, text_bert, wav32k, ssl_content)
+
+    savepath = args.savepath
+    logger.info(f'saved at : {savepath}')
+
+    soundfile.write(savepath, a, vits_hps_data_sampling_rate)
+
+    logger.info('Script finished successfully.')
+
+
+def main():
+    # model files check and download
+    check_and_download_models(WEIGHT_PATH_SSL, MODEL_PATH_SSL, WEIGHT_PATH_SSL)
```

Comment:
@kyakuno 

need to make the following changes.

```suggestion
    check_and_download_models(WEIGHT_PATH_SSL, MODEL_PATH_SSL, REMOTE_PATH)
```

Reply:
2e185f0d5760291f5f3f873525b7660550dd9b3c で解決済み。